sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/query.sql

```diff
@@ -0,0 +1,682 @@
+    ANY_VALUE(distribution_id HAVING MIN submission_date) AS distribution_id,
+    CAST(NULL AS STRING) AS partner_distribution_version,
+    CAST(NULL AS STRING) AS partner_distributor,
+    CAST(NULL AS STRING) AS partner_distributor_channel,
+    CAST(NULL AS STRING) AS partner_id,
+    ANY_VALUE(attribution.campaign HAVING MIN submission_date) AS attribution_campaign,
+    ANY_VALUE(attribution.content HAVING MIN submission_date) AS attribution_content,
+    ANY_VALUE(attribution.experiment HAVING MIN submission_date) AS attribution_experiment,
+    ANY_VALUE(attribution.medium HAVING MIN submission_date) AS attribution_medium,
+    ANY_VALUE(attribution.source HAVING MIN submission_date) AS attribution_source,
+    CAST(NULL AS STRING) AS attribution_ua,
+    ANY_VALUE(
+      default_search_engine_data_load_path
+      HAVING
+        MIN submission_date
+    ) AS engine_data_load_path,
+    ANY_VALUE(default_search_engine_data_name HAVING MIN submission_date) AS engine_data_name,
+    ANY_VALUE(default_search_engine_data_origin HAVING MIN submission_date) AS engine_data_origin,
+    ANY_VALUE(
+      default_search_engine_data_submission_url
+      HAVING
+        MIN submission_date
+    ) AS engine_data_submission_url,
+    CAST(NULL AS STRING) AS apple_model_id,
+    ANY_VALUE(city HAVING MIN submission_date) AS city,
+    CAST(NULL AS STRING) AS db_version,
+    ANY_VALUE(geo_subdivision1 HAVING MIN submission_date) AS subdivision1,
+    ANY_VALUE(normalized_channel HAVING MIN submission_date) AS normalized_channel,
+    ANY_VALUE(country HAVING MIN submission_date) AS country,
+    ANY_VALUE(os HAVING MIN submission_date) AS normalized_os,
+    ANY_VALUE(normalized_os_version HAVING MIN submission_date) AS normalized_os_version,
+    CAST(NULL AS STRING) AS startup_profile_selection_reason,
+    ANY_VALUE(attribution.dltoken HAVING MIN submission_date) AS attribution_dltoken,
+    CAST(NULL AS STRING) AS attribution_dlsource
+  FROM
+    `moz-fx-data-shared-prod.telemetry_derived.clients_daily_v6`
+  WHERE
+    {% if is_init() %}
+      submission_date >= '2010-01-01'
+      {% if parallel_run() %}
+        AND sample_id = {sample_id}
+      {% endif %}
+    {% else %}
+      submission_date = @submission_date
+    {% endif %}
+  GROUP BY
+    client_id,
+    sample_id
+),
+-- The next subquery unions the client's 'data from the 3 pings, resulting in
+-- (max.) 3 records per client_id (one per ping).
+unioned AS (
+  SELECT
+    *,
+    'new_profile' AS ping_source
+  FROM
+    new_profile_ping
+  UNION ALL
+  SELECT
+    *,
+    'shutdown' AS ping_source
+  FROM
+    shutdown_ping
+  UNION ALL
+  SELECT
+    *,
+    'main' AS ping_source
+  FROM
+    main_ping
+),
+-- The following 3 subqueries are required to calculate the second seen date.
+-- First, return one row per client_id with an array of first_seen TIMESTAMPS and
+-- an array of second_seen TIMESTAMPS (one date per reported ping).
+unioned_with_all_dates AS (
```

Comment:
If we decide that only main pings can contribute the second_seen_date, is this unioning still necessary? Could we just aggregate the main pings alone, and take the second date (if it exists) from there?

Reply:
True, this union is not required, I changed it. Just a note, we still need to look at all the pings because not always the 2nd main ping is the second_seen_date of the client. The 1st main ping reported could also be a second_seen_date if another ping was reported on a date before.